Bug 1886262 - Add autograph_xpi signing to adhoc-signing

ahal opened the pull request.

jcristau commented on signingscript/docker.d/passwords.yml:
```diff
@@ -200,6 +200,11 @@ in:
+          - ["https://autograph-external.prod.autograph.services.mozaws.net",
+             {"$eval": "AUTOGRAPH_XPI_USERNAME"},
+             {"$eval": "AUTOGRAPH_XPI_PASSWORD"},
+             ["autograph_xpi", "autograph_xpi_sha1_es256_es384", "autograph_xpi_sha1_es256_ps256", "autograph_xpi_sha1_es256", "autograph_xpi_sha1_ps256"],
```
If we use the same formats here as in the stage case below, the latter will not be reachable (see `signingscript.sign.get_autograph_config`).  To have autograph prod and stage xpi signing in adhoc-t, they should get distinct sets of formats.

ahal replied:
Good catch!

It seems like this function should probably factor in the `keyid` right? Otherwise any formats that the `prod` config has will be unreachable in the `stage` config. If the formats are different, then at least there's *a* format we can test on stage, but we might want to test all of them.

jcristau replied:
I'm not sure.  One thing we could do is have `autograph_stage_xpi*` for stage autograph, and `autograph_xpi*` format using prod.  That'd mirror the existing `autograph_stage_mar384` vs `autograph_mar384` formats.

ahal replied:
Yeah, I like that. Nice and simple

jcristau commented on signingscript/docker.d/passwords.yml:
```diff
@@ -446,6 +452,18 @@ in:
+             {"$eval": "AUTOGRAPH_XPI_PASSWORD"},
+             ["privileged_webextension"],
+             "extension_rsa"  # privileged xpi signing
+            ]
```
Let's be careful not to abuse this, the mozilla-extensions pipeline should be used instead generally.

jcristau commented on signingscript/src/signingscript/sign.py:
```diff
@@ -966,6 +966,9 @@ def _xpi_signing_options(fmt):
             raise SigningScriptError(f"Unsupported format {fmt}")
+    elif fmt == "webextension":
+        cose_algorithms = ["ES256"]
+        digest = "SHA1"
```
This should presumably switch to sha2 eventually?

ahal replied:
I assume so, but this matches what AMO currently does, so willdurand and lgreco wanted to be consistent with that.

willdurand replied:
Yeah, it'd be better to keep `sha1` for now.

jcristau reviewed (approved):
Thanks!

ahal committed:
Bug 1886262 - [signing] Switch xpi and mar variables to new naming format

For consistency, I've made the following changes:

1. Credentials for Autograph stage now go into variables prefixed with `AUTOGRAPH_STAGE`
2. Credentials for privileged xpi signing, now goes into a `XPI_PRIVILEGED` variables

ahal committed:
Bug 1886262 - Add autograph_xpi signing to adhoc-signing

This will give adhoc-signing the ability to sign unprivileged
webextensions.
